solutions/USACO open16-closing-the-farm.mdx

```diff
@@ -0,0 +1,95 @@
+---
+id: usaco-644
+source: USACO Silver 2016 US Open
+title: Closing the Farm
+author: Jesse Choe
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_closing_silver_open16.html)
+
+<LanguageSection>
+<CPPSection>
+
+In this problem, we are asked to determine if all remaining barns are connected as the barns close one at a time. To check this, we can run a depth-first search (DFS) starting from the barn that will close last. To simulate the closure each barn, we can store a boolean array and set each barn in the iteration to be "closed". Then, we can run the DFS and check how many nodes have been visited. If all the unclosed nodes have been visited, then we print "YES"; otherwise, we print "NO".
```

Comment:
```suggestion
In this problem, we are asked to determine if all remaining barns are connected as the barns close one at a time. To check this, we can run a depth-first search (DFS) starting from the barn that will close last. To simulate the closure of each barn, we can store a boolean array and set each barn in the iteration to be "closed". Then, we can run the DFS and check how many nodes have been visited. If all the unclosed nodes have been visited, then we print "YES"; otherwise, we print "NO".
```

Reply:
Yeah, I fixed it. Can you review my new PR now? @frodakcin

Reply:
You should re-request review. Also, do you prefer to stick with long longs?

Reply:
How do I re-request? Since I changed to int's, I don't really care tbh.

Reply:
Top right. Click the button that looks like a cyclic arrow loop thing